hack/tools/protoc-gen-jsonschema/module/enum.go

```diff
@@ -21,9 +21,10 @@ func (m *Module) defineEnum(enum pgs.Enum) *jsonschema.StringSchema {
 
 func (m *Module) schemaForEnum(enum pgs.Enum, rules *validate.EnumRules) (jsonschema.Schema, bool) {
 	required := false
-	schemas := []jsonschema.NonTrivialSchema{m.enumRef(enum)}
```

Comment:
I think this needs to be here. At the moment, we have that any validation rules are additional to the base rule - which sets that the value can only be one of the known enum values.

If we remove this, it does simplify the `const` and `in` rules, but it means that the `not_in` case is no longer correct (it only validates that it's not one of the specified values, but does not validate that it is still one of the known enum values).

We could add the `enumRef` just to the `not_in` branch if we wanted to simplify the `const` and `in` cases - in which case I would say we should just `return` directly from each `if` statement to save checking `len(schemas)`, because these rules are all mutually exclusive.